src/ui/PackageDependencyProvider.ts

```diff
@@ -150,89 +150,59 @@ export class PackageDependenciesProvider implements vscode.TreeDataProvider<Tree
-            const children = [
-                ...this.getLocalDependencies(folderContext),
-                ...this.getRemoteDependencies(folderContext),
-            ];
-            const editedChildren = await this.getEditedDependencies(folderContext);
-            const uneditedChildren: PackageNode[] = [];
-            for (const child of children) {
-                const editedVersion = editedChildren.find(item => item.name === child.name);
-                if (!editedVersion) {
-                    uneditedChildren.push(child);
-                }
-            }
-            return [...uneditedChildren, ...editedChildren].sort((first, second) =>
-                first.name.localeCompare(second.name)
-            );
+            const children = await this.getAllDependencies(folderContext);
+            return children;
         }
 
-        const buildDirectory = buildDirectoryFromWorkspacePath(folderContext.folder.fsPath, true);
-
         if (element instanceof PackageNode) {
             // Read the contents of a package.
-            const packagePath =
-                element.type === "remote"
-                    ? path.join(buildDirectory, "checkouts", getRepositoryName(element.path))
-                    : element.path;
-            return this.getNodesInDirectory(packagePath);
+            return this.getNodesInDirectory(element.path);
         } else {
             // Read the contents of a directory within a package.
             return this.getNodesInDirectory(element.path);
         }
     }
 
-    /**
-     * Returns a {@link PackageNode} for every local dependency
-     * declared in **Package.swift**.
-     */
-    private getLocalDependencies(folderContext: FolderContext): PackageNode[] {
-        const swiftVersion = folderContext.workspaceContext.toolchain.swiftVersion;
-        // prior to Swift 5.6 local dependencies had no requirements
-        if (swiftVersion.isLessThan(new Version(5, 6, 0))) {
-            return folderContext.swiftPackage.dependencies
-                .filter(dependency => !dependency.requirement && dependency.url)
-                .map(
-                    dependency =>
-                        new PackageNode(dependency.identity, dependency.url!, "local", "local")
-                );
-        } else {
-            // since Swift 5.6 local dependencies have `type` `fileSystem`
-            return folderContext.swiftPackage.dependencies
-                .filter(dependency => dependency.type === "fileSystem" && dependency.path)
-                .map(
-                    dependency =>
-                        new PackageNode(dependency.identity, dependency.path!, "local", "local")
+    private async getAllDependencies(folderContext: FolderContext): Promise<PackageNode[]> {
+        return (await folderContext.getWorkspaceDependencies()).map(dependency => {
+            const version =
+                dependency.packageRef.kind === "fileSystem"
```

Comment:
for showing version or branch

Reply:
tested with 5.5, let me check earlier version

Reply:
5.6, 5.5, 5.4
✅ dependency list OK 
✅ edit OK 
✅ unedit OK 

just kept watching `workspace-state.json` changes, it's most relevant to the UI of dependency
* When reset package, it will be deleted, package.resolved has no change
* when edit/unedit package, it will be changed, the path can be read for get the repo, the editing package info will be added into `pins`

probably watching the workspace-state.json is the best choice

Reply:
See my comment above about watching workspace-state.json. The FileSystemWatcher update triggers on every action we run regardless of whether the workspace-state.json changes.

Reply:
`revolve`  will trigger `workspace-state.json` change once even there is no actual change